src/lit-html.ts

```diff
@@ -738,7 +744,19 @@ export class TemplateInstance {
           index++;
           walker.nextNode();
         }
-        this._parts.push(this._partCallback(this, part, walker.currentNode));
+        let node = walker.currentNode;
+        // Swap the comment placeholders for empty text nodes
+        // The nextSibling of the part placeholder is the part endNode
+        if (node.nextSibling !== null && node.nextSibling!.nodeType === 8) {
+          node.parentNode!.replaceChild(document.createTextNode(''), node.nextSibling!);
+        }
+        // The part placeholder is the part startNode
+        if (node.nodeType === 8) {
```

Comment:
Throughout the library we tend to put `/* Node.COMMENT_NODE */` comments behind these magic numbers.
Maybe it'd be nice to look into a `const enum`? That results in the magic numbers in javascript but it keeps them out of the typescript.

```ts
const enum NodeType {
  COMMENT_NODE = 8
}

if (node.nodeType === NodeType.COMMENT_NODE) {
  // foo
}
```

becomes

```js
if (node.nodeType === 8 /* COMMENT_NODE */) {
    // foo
}
```